### PR TITLE
Added feature to sensors to allow json output.

### PR DESCRIPTION
**Pull request by shadethedaemon**
* Added -j option to sensors that outputs json.
* -j option overrides raw option

### CODE REVIEW

File: prog/sensors/main.c [hunk 1]
```diff
@@ -61,6 +61,7 @@ static void print_long_help(void)
 	     "  -A, --no-adapter      Do not show adapter for each chip\n"
 	     "      --bus-list        Generate bus statements for sensors.conf\n"
 	     "  -u                    Raw output\n"
+			 "  -j                    Json output\n"
```
Comment (groeck) [inline prog/sensors/main.c]: Alignment s off

File: prog/sensors/sensors.1 [hunk 2]
```diff
@@ -10,15 +10,15 @@
 .\" manual under the conditions for verbatim copying, provided that the
 .\" entire resulting derived work is distributed under the terms of a
 .\" permission notice identical to this one
-.\"
```
Comment (groeck) [inline prog/sensors/sensors.1]: Any idea what changed here ? If possible please undo; this is an unrelated change.

File: prog/sensors/sensors.1 [hunk 3]
```diff
@@ -10,15 +10,15 @@
...
-.\" 
+.\"
 .\" Since the Linux kernel and libraries are constantly changing, this
 .\" manual page may be incorrect or out-of-date.  The author(s) assume no
 .\" responsibility for errors or omissions, or for damages resulting from
 .\" the use of the information contained herein.  The author(s) may not
 .\" have taken the same level of care in the production of this manual,
 .\" which is licensed free of charge, as they might when working
 .\" professionally.
-.\"
```
Comment (groeck) [inline prog/sensors/sensors.1]: same here

File: prog/sensors/sensors.1 [hunk 4]
```diff
@@ -66,6 +66,8 @@ Raw output. This mode is suitable for debugging and for post-processing
 of the output by scripts. It is also useful when writing a configuration
 file because it shows the raw input names which must be referenced in the
 configuration file.
+.IP -j
+Raw output. This mode is suitable for post-processing of the output by scripts.
```
Comment (groeck) [inline prog/sensors/sensors.1]: Raw ?

File: prog/sensors/main.c [hunk 5]
```diff
@@ -204,15 +219,25 @@ static int do_the_real_work(const sensors_chip_name *match, int *err)
...
 			if (do_a_set(chip))
 				*err = 1;
 		} else
-			do_a_print(chip);
+			if (do_json) {
+				if (cnt > 0)
+					printf(",\n");
+				do_a_json_print(chip);
+			}
+			else
```
Comment (groeck) [inline prog/sensors/main.c]: Please use { } in both branches of the if statement for readability/consistency, and put "} else {" on a single line for consistency with the rest of the code.

File: prog/sensors/chips.c [hunk 6]
```diff
@@ -67,6 +67,53 @@ void print_chip_raw(const sensors_chip_name *name)
...
+					if (subCnt > 0)
+						printf(",\n", label);
+					printf("         \"%s\": %.3f", sub->name, val);
+				}
+
+			} else
+			printf("(%s)", label);
+			subCnt++;
+		}
+		printf("\n      }", label);
```
Comment (groeck) [inline prog/sensors/chips.c]: Not sure what you want to do here (print the label or not ?), but as written this is wrong (and gives you a warning when compiling).

File: prog/sensors/chips.c [hunk 7]
```diff
@@ -67,6 +67,53 @@ void print_chip_raw(const sensors_chip_name *name)
...
+			if (sub->flags & SENSORS_MODE_R) {
+				if ((err = sensors_get_value(name, sub->number,
+							     &val)))
+					fprintf(stderr, "ERROR: Can't get "
+						"value of subfeature %s: %s\n",
+						sub->name,
+						sensors_strerror(err));
+				else {
+					if (subCnt > 0)
+						printf(",\n", label);
```
Comment (groeck) [inline prog/sensors/chips.c]: Wrong

File: prog/sensors/chips.c [hunk 8]
```diff
@@ -67,6 +67,53 @@ void print_chip_raw(const sensors_chip_name *name)
...
+	a = 0;
+	cnt = 0;
+	while ((feature = sensors_get_features(name, &a))) {
+		if (!(label = sensors_get_label(name, feature))) {
+			fprintf(stderr, "ERROR: Can't get label of feature "
+				"%s!\n", feature->name);
+			continue;
+		}
+		if (cnt > 0)
+			printf(",\n", label);
```
Comment (groeck) [inline prog/sensors/chips.c]: This is wrong.

File: prog/sensors/chips.c [hunk 9]
```diff
@@ -67,6 +67,53 @@ void print_chip_raw(const sensors_chip_name *name)
...
+				}
+
+			} else
+			printf("(%s)", label);
+			subCnt++;
+		}
+		printf("\n      }", label);
+		cnt++;
+	}
+	printf("\n", label);
```
Comment (groeck) [inline prog/sensors/chips.c]: Wrong

File: prog/sensors/chips.c [hunk 10]
```diff
@@ -67,6 +67,53 @@ void print_chip_raw(const sensors_chip_name *name)
...
+						sub->name,
+						sensors_strerror(err));
+				else {
+					if (subCnt > 0)
+						printf(",\n", label);
+					printf("         \"%s\": %.3f", sub->name, val);
+				}
+
+			} else
+			printf("(%s)", label);
```
Comment (groeck) [inline prog/sensors/chips.c]: alignment is wrong

File: prog/sensors/sensors.1 (hunk 2 above)
Comment (shadethedaemon) [inline prog/sensors/sensors.1]: It looked like there was a space that got removed, I'll add it back.

**Comment by shadethedaemon** [comment]
I have made the corrections that were requested, I see i forgot to remove ", label". I have two ways of pushing the changes. I can add another commit which will automatically get added to this pull request. I can also close this pull request, amend the changes to the commit and submit a new pull request with only one commit.

**Comment by groeck** [comment]
The latter (new pull request with single commit) is always better, since otherwise I'll have to merge the two patches myself. Also, it would be great if you can sign your patches.
